tuftool/src/update.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::build_targets;
+use crate::datetime::parse_datetime;
+use crate::error::{self, Result};
+use crate::source::parse_key_source;
+use chrono::{DateTime, Utc};
+use snafu::ResultExt;
+use std::fs::File;
+use std::num::{NonZeroU64, NonZeroUsize};
+use std::path::PathBuf;
+use structopt::StructOpt;
+use tough::editor::RepositoryEditor;
+use tough::key_source::KeySource;
+use tough::{ExpirationEnforcement, FilesystemTransport, HttpTransport, Limits, Repository};
+use url::Url;
+
+#[derive(Debug, StructOpt)]
+pub(crate) struct UpdateArgs {
+    /// Follow symbolic links in `indir`
+    #[structopt(short = "f", long = "follow")]
+    follow: bool,
+
+    /// Number of target hashing threads to run (default: number of cores)
+    // No default is specified in structopt here. This is because rayon
+    // automatically spawns the same number of threads as cores when any
+    // of its parallel methods are called.
+    #[structopt(short = "j", long = "jobs")]
+    jobs: Option<NonZeroUsize>,
+
+    /// Key files to sign with
+    #[structopt(short = "k", long = "key", required = true, parse(try_from_str = parse_key_source))]
+    keys: Vec<Box<dyn KeySource>>,
+
+    /// Version of snapshot.json file
+    #[structopt(long = "snapshot-version")]
+    snapshot_version: NonZeroU64,
+    /// Expiration of snapshot.json file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(long = "snapshot-expires", parse(try_from_str = parse_datetime))]
+    snapshot_expires: DateTime<Utc>,
+
+    /// Version of targets.json file
+    #[structopt(long = "targets-version")]
+    targets_version: NonZeroU64,
+    /// Expiration of targets.json file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(long = "targets-expires", parse(try_from_str = parse_datetime))]
+    targets_expires: DateTime<Utc>,
+
+    /// Version of timestamp.json file
+    #[structopt(long = "timestamp-version")]
+    timestamp_version: NonZeroU64,
+    /// Expiration of timestamp.json file; can be in full RFC 3339 format, or something like 'in
+    /// 7 days'
+    #[structopt(long = "timestamp-expires", parse(try_from_str = parse_datetime))]
+    timestamp_expires: DateTime<Utc>,
+
+    /// Path to root.json file for the repository
+    #[structopt(short = "r", long = "root")]
+    root: PathBuf,
+
+    /// TUF repository metadata base URL
+    #[structopt(short = "m", long = "metadata-url")]
+    metadata_base_url: Url,
+
+    /// The working directory where the current metadata files will be written.
+    #[structopt(short = "w", long = "workdir", default_value = ".")]
+    workdir: PathBuf,
```

Comment:
It looks like this is just a temporary location for tough's data store?  Can we use a tempdir instead of making the user specify?

Reply:
I could see situations where a user may want to inspect those files as they were.

How about keeping the argument, but using a tempdir if the user *doesn't* pass the arg?

Reply:
I think wanting to inspect those files is... odd?  They can inspect the original repo and the resulting repo; what's left to care about?

If you think it's important, how about clarifying the flag name to show that it persists the data store to a given path?  It's just a rename of workdir I think - `--persist-tough-data-store PATH` or something?

Reply:
True - I guess a user can just inspect that actual repo. I'll remove the arg.